frame/support/procedural/src/lib.rs

```diff
@@ -583,3 +584,646 @@ pub fn storage_alias(_: TokenStream, input: TokenStream) -> TokenStream {
+///
+/// If `#[pallet::generate_deposit]` is present then the macro implements `fn deposit_event` on
+/// `Pallet`.
+#[proc_macro_attribute]
+pub fn generate_deposit(_: TokenStream, _: TokenStream) -> TokenStream {
+	pallet_macro_stub()
+}
+
+/// The `#[pallet::storage]` attribute lets you define some abstract storage inside of runtime
+/// storage and also set its metadata. This attribute can be used multiple times.
+///
+/// Item should be defined as:
+///
+/// ```ignore
+/// #[pallet::storage]
+/// #[pallet::getter(fn $getter_name)] // optional
+/// $vis type $StorageName<$some_generic> $optional_where_clause
+/// 	= $StorageType<$generic_name = $some_generics, $other_name = $some_other, ...>;
+/// ```
+///
+/// or with unnamed generic:
+///
+/// ```ignore
+/// #[pallet::storage]
+/// #[pallet::getter(fn $getter_name)] // optional
+/// $vis type $StorageName<$some_generic> $optional_where_clause
+/// 	= $StorageType<_, $some_generics, ...>;
+/// ```
+///
+/// I.e. it must be a type alias, with generics: `T` or `T: Config`, aliased type must be one
+/// of `StorageValue`, `StorageMap` or `StorageDoubleMap`. The generic arguments of the storage
+/// type can be given in two manners: named and unnamed. For named generic arguments, the name
+/// for each argument should match the name defined for it on the storage struct:
+/// * `StorageValue` expect `Value` and optionally `QueryKind` and `OnEmpty`,
+/// * `StorageMap` expect `Hasher`, `Key`, `Value` and optionally `QueryKind` and `OnEmpty`,
+/// * `CountedStorageMap` expect `Hasher`, `Key`, `Value` and optionally `QueryKind` and `OnEmpty`,
+/// * `StorageDoubleMap` expect `Hasher1`, `Key1`, `Hasher2`, `Key2`, `Value` and optionally
+///   `QueryKind` and `OnEmpty`.
+///
+/// For unnamed generic arguments: Their first generic must be `_` as it is replaced by the
+/// macro and other generic must declared as a normal generic type declaration.
+///
+/// The `Prefix` generic written by the macro is generated using
+/// `PalletInfo::name::<Pallet<..>>()` and the name of the storage type. E.g. if runtime names
+/// the pallet "MyExample" then the storage `type Foo<T> = ...` should use the prefix:
+/// `Twox128(b"MyExample") ++ Twox128(b"Foo")`.
+///
+/// For the `CountedStorageMap` variant, the `Prefix` also implements
+/// `CountedStorageMapInstance`. It also associates a `CounterPrefix`, which is implemented the
+/// same as above, but the storage prefix is prepend with `"CounterFor"`. E.g. if runtime names
+/// the pallet "MyExample" then the storage `type Foo<T> = CountedStorageaMap<...>` will store
+/// its counter at the prefix: `Twox128(b"MyExample") ++ Twox128(b"CounterForFoo")`.
+///
+/// E.g:
+///
+/// ```ignore
+/// #[pallet::storage]
+/// pub(super) type MyStorage<T> = StorageMap<Hasher = Blake2_128Concat, Key = u32, Value = u32>;
+/// ```
+///
+/// In this case the final prefix used by the map is `Twox128(b"MyExample") ++
+/// Twox128(b"OtherName")`.
```

Comment:
Nit: is `++` a thing in rust?

Reply:
Rust genuinely does not have `++`, so should just be `+`?

Reply:
This just means concatenation.

Reply:
should we maybe do this to be clearer then?
```rust
/// `concat(Twox128(b"MyExample"), Twox128(b"OtherName"))`
```

Reply:
This is used in several places in the docs.

Reply:
ok as long as people know that ++ means concat. I wasn't aware of that one but happy to leave it!